common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -450,7 +461,7 @@ public static UTF8String toTitleCase(final UTF8String target, final int collatio
     return UTF8String.fromString(toTitleCase(target.toString(), collationId));
   }
 
-  public static String toTitleCase(final String target, final int collationId) {
+  private static String toTitleCase(final String target, final int collationId) {
```

Comment:
Same concerns/question as for the other overload of `toTitleCase`.

Reply:
discussing in: https://github.com/apache/spark/pull/47014#discussion_r1652818328